front/components/agent_builder/capabilities/mcp/MCPServerViewsDialog.tsx

```diff
@@ -0,0 +1,694 @@
+          selectedToolsInDialog={selectedToolsInDialog}
+          onRemoveSelectedTool={toggleToolSelection}
+        />
+      ),
+    },
+    {
+      id: CONFIGURATION_DIALOG_PAGE_IDS.CONFIGURATION,
+      title: mcpServerView?.name || "Configure Tool",
+      description: "",
+      icon: undefined,
+      content:
+        configurationTool && mcpServerView && requirements && formSchema ? (
+          <FormProvider form={form}>
+            <div>
+              <div className="space-y-6">
+                <MCPActionHeader
+                  action={configurationTool}
+                  mcpServerView={mcpServerView}
+                  allowNameEdit={!configurationTool.noConfigurationRequired}
+                />
+
+                {requirements.requiresReasoningConfiguration && (
+                  <ReasoningModelSection owner={owner} />
+                )}
+
+                {requirements.requiresChildAgentConfiguration && (
+                  <ChildAgentSection owner={owner} />
+                )}
+
+                {requirements.mayRequireTimeFrameConfiguration && (
+                  <TimeFrameSection actionType="search" />
+                )}
+
+                {requirements.mayRequireJsonSchemaConfiguration && (
+                  <JsonSchemaSection
+                    owner={owner}
+                    fieldName="configuration.jsonSchema"
+                  />
+                )}
+              </div>
+            </div>
+          </FormProvider>
+        ) : (
+          <div className="flex h-40 w-full items-center justify-center">
+            <Spinner />
+          </div>
+        ),
+    },
+    {
+      id: CONFIGURATION_DIALOG_PAGE_IDS.INFO,
+      title: infoMCPServerView?.name || "Tool Information",
+      description: "",
+      icon: undefined,
+      content: infoMCPServerView ? (
+        <div className="flex h-full flex-col space-y-2">
+          <div className="flex items-center space-x-2 text-base font-medium">
+            {getAvatarFromIcon(infoMCPServerView.server.icon, "sm")}
+            <span className="">{infoMCPServerView.server.name}</span>
+          </div>
+          <span className="text-sm text-muted-foreground dark:text-muted-foreground-night">
+            {infoMCPServerView.server.description}
+          </span>
+          <ToolsList
+            owner={owner}
+            mcpServerView={infoMCPServerView}
+            disableUpdates
+          />
+        </div>
+      ) : (
+        <div className="flex h-40 w-full items-center justify-center">
+          <Spinner />
+        </div>
+      ),
+    },
+  ];
+
+  const handleCancel = () => {
+    setIsOpen(false);
+    onModeChange(null);
+    if (isAddMode) {
+      setSelectedToolsInDialog([]);
+      setConfigurationTool(null);
+      setConfigurationMCPServerView(null);
+      setCurrentPageId(CONFIGURATION_DIALOG_PAGE_IDS.TOOL_SELECTION);
+    }
+  };
+
+  const handleConfigurationSave = useCallback(async () => {
```

Comment:
You should have access to the form here, made an other suggestion below to make it work.
```suggestion
  const handleConfigurationSave = useCallback(async (data: MCPFormData) => {
```
This would avoid having to `await form.trigger()` and `form.getValues()`, as it should be handle by the react-hook-form
This might require some changes on how to display the error message. Or was that on purpose?